subsys/net/lib/mqtt/mqtt.c

```diff
@@ -8,6 +8,7 @@
 #include "mqtt_pkt.h"
 
 #include <net/net_ip.h>
+#include <net/zstream.h>
 #include <net/async_socket.h>
```

Comment:
Then, if we are going to go ahead with this stream API, I would want to rename async_socket.h module to async_zstream_sock.h,  to be clear that we are not using sockets directly.